changelog/stdcpp.dd

```diff
@@ -0,0 +1,64 @@
+Ddoc
+
+$(H2 Transition to C++11 character types)
+  
+          With C++11 comes the advent of changed character type mangling.
+          D's default behavior will be to conform to this after a one 
+          release transition period. A new switch -stdc++={c++98,c++11} is 
+          added to control the version that compatibility is set to.
+          This switch sets `__traits(getTargetInfo, "cppStd")` to the value of 
+          `__cplusplus` that the corresponding version of the C++ standard defines.
+		  This will be available as `__cplusplus` from DRuntime's C++ bindings.
```

Comment:
Is it necessary to create a `__cplusplus` symbol as well when we have `getTargetInfo`? It can't be that common for end users to need to access this information. For implementing the actual bindings I think it's fine to use `getTargetInfo`.

Reply:
It is done for the runtime library https://github.com/dlang/druntime/blob/master/src/core/stdcpp/xutility.d#L50

Reply:
I don't see why that needs to be public.

Reply:
Eh, there's no harm, and it makes translating `#if __cplusplus >= whatever` obvious and straight forward. I expect it will come with an enum for the various magic numbers.

Reply:
Well, one of the points of implementing `getTargetInfo` as a trait is to avoid adding a bunch of new magic symbols. But if we're going to make everything available as a public symbol anyway, it's almost the same.

I prefer that we create a new module, `core.target_info`, which contains something like a struct `TargetInfo` which collects all this information in a nice API.

Reply:
Its just a regular symbol defined through a trait. Doing it as a trait is the easiest way and `getTargetInfo` was made to avoid special tokens.

Reply:
Yes, I know, I even suggested it to be implemented as a trait or similar from the beginning.

`TargetInfo` would be instead of `__cplusplus`. `TargetInfo` would get the data from `__traits(getTargetInfo)`.

Reply:
Using a `TargetInfo` would mean pulling in stuff that is possibly not used, no to mention that the availability of info may be dependant on compiler/flags/available features so that would be a lot of complicated logic for not particularly much gain.

It is also very clear what `__cplusplus` would mean as the direct equivalent in c++.

Reply:
> Using a TargetInfo would mean pulling in stuff that is possibly not used

If that's a problem, just use `__traits(getTargetInfo)`. For the implementation of the bindings I definitely think we shouldn't add any extra public symbols and we don't need a potential `TargetInfo` either. If the user wants to have a convenience symbol like `__cplusplus` then the user can add that to his/her own code.

Reply:
`__cplusplus` with two leading underscores is reserved. It is much easier to type, has identical meaning in C++, is going to be used in the C++ bindings and the windows API. I fail to see why we shouldn't add it.

Reply:
I suggest postpone the decision about the of the icing of the cake, or at least the announcement of it, until later. There's no need to hold up this PR for something that may change in the future.

Reply:
OK, done.